Dockerfile

```diff
@@ -0,0 +1,31 @@
+# select build image
+FROM rust as builder
+
+# create a new empty shell project
+RUN USER=root cargo new --bin trin
+WORKDIR /trin
+
+RUN apt-get update && apt-get install clang -y
+RUN rustup component add rustfmt
+
+# copy over manifests and source to build image
+COPY ./Cargo.lock ./Cargo.lock
+COPY ./Cargo.toml ./Cargo.toml
+COPY ./src ./src 
+COPY ./trin-core ./trin-core 
+COPY ./trin-history ./trin-history 
+COPY ./trin-state ./trin-state 
+COPY ./ethportal-peertest ./ethportal-peertest 
```

Comment:
This is going to bite us roughly every time a new crate is added... I'm wondering if there's any better way, like just copying the whole directory minus `.git`. This isn't a hard blocker, just something to consider.